src/compiler/checker.ts

```diff
@@ -5341,37 +5341,53 @@ namespace ts {
                 if (isArrayType(type)) {
                     return createArrayType(getWidenedType((<TypeReference>type).typeArguments[0]));
                 }
+                if (isTupleType(type)) {
+                    return createTupleType(map(type.elementTypes, getWidenedType));
+                }
             }
             return type;
         }
 
-        function reportWideningErrorsInType(type: Type): boolean {
+        function reportWideningErrorsInType(type: Type, expression: Expression): boolean {

```

Comment:
Additionally, I don't know what `expression` is supposed to represent, nor the context for which you're feeding in type. When will you need to do